exporter/http.go

```diff
@@ -107,3 +115,27 @@ func (e *Exporter) reloadPwdFile(w http.ResponseWriter, r *http.Request) {
 	e.Unlock()
 	_, _ = w.Write([]byte(`ok`))
 }
+
+func (e *Exporter) isBasicAuthConfigured() bool {
+	return e.options.BasicAuthUsername != "" && e.options.BasicAuthPassword != ""
+}
+
+func (e *Exporter) verifyBasicAuth(user, password string, authHeaderSet bool) error {
+
+	if !e.isBasicAuthConfigured() {
+		return nil
+	}
+
+	if !authHeaderSet {
+		return errors.New("Unauthorized")
+	}
+
+	userCorrect := subtle.ConstantTimeCompare([]byte(user), []byte(e.options.BasicAuthUsername))
+	passCorrect := subtle.ConstantTimeCompare([]byte(password), []byte(e.options.BasicAuthPassword))
```

Comment:
nit: looks like you're always using these as `byte[]` instead of string so might as well just move the conversion into the "setter" in main.go:206/207